slidechain/custodian.go

```diff
@@ -106,6 +107,7 @@ func newCustodian(ctx context.Context, db *sql.DB, hclient horizon.ClientInterfa
 		hclient:       hclient,
 		imports:       sync.NewCond(new(sync.Mutex)),
 		exports:       sync.NewCond(new(sync.Mutex)),
+		pegouts:       sync.NewCond(new(sync.Mutex)),
```

Comment:
This was super helpful - I think most of this should be covered now. Within `watchPegOuts`, a go routine polls periodically using `time.NewTicker`, and there's also a loop over values from a channel as you described.

A concurrency problem did arise, and I'm curious if I approached it correctly. The new approach could `SELECT` some rows within each routine. That becomes especially problematic if they're also post-processed twice, since that would attempt to delete an already deleted row. As a result, I defined a new mutex `exportmux` on the `Custodian` to avoid double-processing exported rows. I treated the entire iteration of a for-loop within each routine as a critical section, and the new mutex locks and unlocks accordingly.

I tried to approach this using channels, but they seemed like overkill - a mutex would be sufficient. Would channels have been a better approach?

Reply:
> the post-pegout tx can fail separately from the peg-out tx.

I thought about this as well, but I'm not certain what the right protection or behavior is for this. Currently, if the post-pegout tx fails, `doPegOutTx` will return an error, which will result in a `log.Fatal` in `watchPegOuts`. What behavior would we want instead?